app/helpers/repository_helper.rb

```diff
@@ -161,33 +161,48 @@ def remove_folders(folders, user, repo, path: '/', txn: nil)
     end
 
     current_path = Pathname.new path
+    relative_path = Pathname.new(folders[0])
+    abs_path = File.join(current_path, relative_path)
+
+    return [true, [:not_exist]] unless repo.get_latest_revision.path_exists?(abs_path)
 
     current_revision = repo.get_latest_revision.revision_identifier
 
-    dirs = {}
+    dirs = []
     files = []
     folders.each do |folder_path|
       folder_path = current_path.join(folder_path).to_s
       next if dirs.include? folder_path
 
-      repo.get_revision(current_revision).tree_at_path(folder_path, with_attrs: false).each do |_, obj|
+      repo.get_revision(current_revision.to_s).tree_at_path(folder_path, with_attrs: false).each do |_, obj|
         path = File.join obj.path, obj.name
         if obj.is_a? Repository::RevisionFile
           files << path
         else
-          dirs[path] = current_revision
+          dirs << path
         end
       end
-      dirs[folder_path] = current_revision
+      dirs = dirs.reverse
+      dirs << folder_path
+    end
+    unless files.empty?
+      success, file_messages = remove_files(files, user, repo, path: '', txn: txn, keep_folder: false)
+      return [success, file_messages] unless success
     end
-    success, file_messages = remove_files(files, user, repo, path: '', txn: txn)
 
-    return [success, file_messages] unless success
+    dirs.each do |dir|
+      txn.remove_directory(dir, current_revision.to_s)
+    end
 
-    # folders are removed in git if their contents are removed
-    unless repo.is_a? GitRepository
-      dirs.each do |dir, revision|
-        txn.remove(dir, revision)
+    if repo.is_a? GitRepository
+      if relative_path.dirname == Pathname.new('.')
+        unless repo.get_latest_revision.path_exists?(File.join(current_path, '.gitkeep'))
```

Comment:
You should be able to access the constant name (`.gitkeep`) rather than hard-coding it again here.

Reply:
I see you changed it in the two lines below, which is good, but you should also change the occurrence of `.gitkeep` on this line.

Also, rather than use `internal_file_names`, please use `DUMMY_FILE_NAME` instead.